dsn.go

```diff
@@ -575,3 +578,11 @@ func parseDSNParams(cfg *Config, params string) (err error) {
 
 	return
 }
+
+func ensureHavePort(addr string) string {
+	s := strings.Split(addr, ":")
+	if len(s) == 1 {
```

Comment:
`strings.ContainsRune(addr, ':')`

Reply:
Or use [SplitHostPort](https://golang.org/pkg/net/#SplitHostPort) to support IPv6

Reply:
```
if _, _, err := net.SplitHostPort(addr); err != nil {
    addr = net.JoinHostPort(addr, "3306")
}
return addr
```

Reply:
@methane , thank you, good approach.

Reply:
updated.